### PR TITLE
chore: shake core

### PR DESCRIPTION
**Pull request by Kha**

### CODE REVIEW

**Comment by Kha** [comment]
!bench

**Comment by leanprover-radar** [comment]
[Benchmark results](https://radar.lean-lang.org/repos/lean4/commits/d901f24f081b630af7b4aa3bd28487be7bb88008?reference=d1514f3cecc4139039962137fcf3a84c9459194a) for d901f24f081b630af7b4aa3bd28487be7bb88008 against d1514f3cecc4139039962137fcf3a84c9459194a are in! @Kha

- 🟥 `build` exited with code **2**
- 🟥 `other` exited with code **2**

No significant changes detected.

**Comment by Kha** [comment]
!bench

**Comment by leanprover-radar** [comment]
[Benchmark results](https://radar.lean-lang.org/repos/lean4/commits/c8a94a2d9bc36d129839cc3972a517c26fddde64?reference=d1514f3cecc4139039962137fcf3a84c9459194a) for c8a94a2d9bc36d129839cc3972a517c26fddde64 against d1514f3cecc4139039962137fcf3a84c9459194a are in! @Kha

- ✅ `build//instructions`: **-282.3G** (**-2.20%**)
- 🟥 `other` exited with code **1**

**Large changes (1✅)**

- ✅ `build/module/Std.Tactic.BVDecide.Bitblast.BVExpr.Circuit.Lemmas.Expr//instructions`: **-10.7G** (**-8.50%**)

**Medium changes (16✅)**

- ✅ `build//instructions`: **-282.3G** (**-2.20%**)
- ✅ `build/module/Init.Data.Array.QSort.Basic//instructions`: **-1.9G** (**-12.46%**) (reduced significance based on absolute threshold)
- ✅ `build/module/Init.Data.String.Decode//instructions`: **-1.1G** (**-3.34%**)
- ✅ `build/module/Lean.Elab.DocString//instructions`: **-1.0G** (**-2.14%**) (reduced significance based on absolute threshold)
- ✅ `build/module/Lean.Elab.MutualInductive//instructions`: **-2.0G** (**-5.13%**) (reduced significance based on absolute threshold)
- ✅ `build/module/Lean.Meta.Injective//instructions`: **-1.5G** (**-9.74%**) (reduced significance based on absolute threshold)
- ✅ `build/module/Lean.Meta.Sym.Pattern//instructions`: **-1.8G** (**-6.29%**) (reduced significance based on absolute threshold)
- ✅ `build/module/Lean.Meta.Sym.ReplaceS//instructions`: **-1.2G** (**-11.68%**) (reduced significance based on absolute threshold)
- ✅ `build/module/Lean.Meta.Tactic.Grind.AC.Seq//instructions`: **-1.2G** (**-14.46%**) (reduced significance based on absolute threshold)
- ✅ `build/module/Std.Data.DTreeMap.Internal.Model//instructions`: **-1.6G** (**-2.33%**)
- ✅ `build/module/Std.Tactic.BVDecide.Bitblast.BVExpr.Circuit.Impl.Expr//instructions`: **-3.4G** (**-4.75%**) (reduced significance based on absolute threshold)
- ✅ `build/module/Std.Tactic.BVDecide.Bitblast.BVExpr.Circuit.Impl.Operations.Mul//instructions`: **-2.0G** (**-7.38%**) (reduced significance based on absolute threshold)
- ✅ `build/module/Std.Tactic.BVDecide.Bitblast.BVExpr.Circuit.Lemmas.Operations.ShiftRight//instructions`: **-1.5G** (**-6.56%**) (reduced significance based on absolute threshold)
- ✅ `build/module/Std.Tactic.BVDecide.LRAT.Internal.Formula.RatAddSound//instructions`: **-1.4G** (**-6.57%**) (reduced significance based on absolute threshold)
- ✅ `build/module/Std.Tactic.BVDecide.LRAT.Internal.Formula.RupAddResult//instructions`: **-2.3G** (**-4.14%**) (reduced significance based on absolute threshold)
- ✅ `build/module/Std.Tactic.BVDecide.LRAT.Internal.Formula.RupAddSound//instructions`: **-2.2G** (**-10.79%**) (reduced significance based on absolute threshold)

and 1 hidden

**Small changes (1813✅, 23🟥)**

Too many entries to display here. View the full report on radar instead.

**Comment by Kha** [comment]
!bench

**Comment by leanprover-radar** [comment]
[Benchmark results](https://radar.lean-lang.org/repos/lean4/commits/2e1ad8623fe67bf13040b7d13902dfb7639e50c3?reference=d1514f3cecc4139039962137fcf3a84c9459194a) for 2e1ad8623fe67bf13040b7d13902dfb7639e50c3 against d1514f3cecc4139039962137fcf3a84c9459194a are in! @Kha

- ✅ `build//instructions`: **-280.8G** (**-2.19%**)
- 🟥 `other` exited with code **1**

**Large changes (1✅)**

- ✅ `build/module/Std.Tactic.BVDecide.Bitblast.BVExpr.Circuit.Lemmas.Expr//instructions`: **-10.7G** (**-8.50%**)

**Medium changes (16✅)**

- ✅ `build//instructions`: **-280.8G** (**-2.19%**)
- ✅ `build/module/Init.Data.Array.QSort.Basic//instructions`: **-1.9G** (**-12.49%**) (reduced significance based on absolute threshold)
- ✅ `build/module/Init.Data.String.Decode//instructions`: **-1.1G** (**-3.36%**)
- ✅ `build/module/Lean.Elab.DocString//instructions`: **-1.0G** (**-2.13%**) (reduced significance based on absolute threshold)
- ✅ `build/module/Lean.Elab.MutualInductive//instructions`: **-2.0G** (**-5.13%**) (reduced significance based on absolute threshold)
- ✅ `build/module/Lean.Meta.Injective//instructions`: **-1.5G** (**-9.74%**) (reduced significance based on absolute threshold)
- ✅ `build/module/Lean.Meta.Sym.Pattern//instructions`: **-1.8G** (**-6.30%**) (reduced significance based on absolute threshold)
- ✅ `build/module/Lean.Meta.Sym.ReplaceS//instructions`: **-1.2G** (**-11.71%**) (reduced significance based on absolute threshold)
- ✅ `build/module/Lean.Meta.Tactic.Grind.AC.Seq//instructions`: **-1.2G** (**-14.46%**) (reduced significance based on absolute threshold)
- ✅ `build/module/Std.Data.DTreeMap.Internal.Model//instructions`: **-1.6G** (**-2.39%**)
- ✅ `build/module/Std.Tactic.BVDecide.Bitblast.BVExpr.Circuit.Impl.Expr//instructions`: **-3.4G** (**-4.77%**) (reduced significance based on absolute threshold)
- ✅ `build/module/Std.Tactic.BVDecide.Bitblast.BVExpr.Circuit.Impl.Operations.Mul//instructions`: **-2.0G** (**-7.37%**) (reduced significance based on absolute threshold)
- ✅ `build/module/Std.Tactic.BVDecide.Bitblast.BVExpr.Circuit.Lemmas.Operations.ShiftRight//instructions`: **-1.5G** (**-6.59%**) (reduced significance based on absolute threshold)
- ✅ `build/module/Std.Tactic.BVDecide.LRAT.Internal.Formula.RatAddSound//instructions`: **-1.4G** (**-6.57%**) (reduced significance based on absolute threshold)
- ✅ `build/module/Std.Tactic.BVDecide.LRAT.Internal.Formula.RupAddResult//instructions`: **-2.3G** (**-4.14%**) (reduced significance based on absolute threshold)
- ✅ `build/module/Std.Tactic.BVDecide.LRAT.Internal.Formula.RupAddSound//instructions`: **-2.2G** (**-10.81%**) (reduced significance based on absolute threshold)

and 1 hidden

**Small changes (1813✅, 23🟥)**

Too many entries to display here. View the full report on radar instead.

File: src/lake/Lake/Config/Opaque.lean [hunk 1]
```diff
@@ -8,6 +8,7 @@ module
 prelude
 public import Init.Prelude
 meta import all Lake.Util.OpaqueType
+import Lake.Util.OpaqueType
```
Comment (tydeu) [inline src/lake/Lake/Config/Opaque.lean:11]: It is interesting that shake is adding private imports to the module that were not previously required for the module to build. What is logic behind this?

File: src/lake/Lake/Util/OpaqueType.lean [hunk 2]
```diff
@@ -6,8 +6,9 @@ Authors: Mac Malone
 module
 
 prelude
-public import Init.Prelude
 public meta import Lake.Util.Binder
+public import Init.Notation
```
Comment (tydeu) [inline src/lake/Lake/Util/OpaqueType.lean:10]: It promoted this import from `Init.Prelude` to `Init.Notation`, why?  I would never have expected shake to potentially *increase* the transitive import closure.

File: src/lake/Lake/Config/Opaque.lean (hunk 1 above)
Comment (Kha) [reply src/lake/Lake/Config/Opaque.lean:11]: I'm not surprised it got confused by `meta import all`. That line doesn't even seem to be needed?

File: src/lake/Lake/Util/OpaqueType.lean (hunk 2 above)
Comment (Kha) [reply src/lake/Lake/Util/OpaqueType.lean:10]: It's already in the `public meta` closure, it doesn't understand this edge case of this sometimes being equivalent

**Comment by Kha** [comment]
@adomani Could you suggest a better way to write the test at https://github.com/leanprover-community/mathlib4-nightly-testing/commit/f62bb72a11cb951eedb1157a6fe64561c7f66be8? Changes to the mere number of syntax declarations in core being able to break Mathlib tests is not a workable solution. /cc @kim-em

**Comment by Kha** [comment]
!bench

**Comment by leanprover-radar** [comment]
[Benchmark results](https://radar.lean-lang.org/repos/lean4/commits/080b41f284d2d0fef1962a0110cd1c9ef7e8309c?reference=35852a9fa9e8b65d0276064cca2ed999a7f6a175) for 080b41f284d2d0fef1962a0110cd1c9ef7e8309c against 35852a9fa9e8b65d0276064cca2ed999a7f6a175 are in! @Kha

- ✅ `build//instructions`: **-842.0M** (**-0.01%**)

**Medium changes (5✅)**

- ✅ `lake config elab//instructions`: **-86.2M** (**-2.94%**)
- ✅ `lake config import//instructions`: **-83.8M** (**-5.61%**)
- ✅ `lake config tree//instructions`: **-82.4M** (**-5.38%**)
- ✅ `lake env//instructions`: **-84.7M** (**-5.59%**)
- ✅ `simp_arith1//instructions`: **-68.0M** (**-2.59%**)

**Small changes (57✅, 42🟥)**

Too many entries to display here. View the full report on radar instead.

**Comment by Kha** [comment]
!bench

**Comment by leanprover-radar** [comment]
[Benchmark results](https://radar.lean-lang.org/repos/lean4/commits/4565de9276cef4ebdd2560290b70c46a86aee671?reference=47a18a4bb44fb88dfa934e1aa3cfcd6a1ff6d047) for 4565de9276cef4ebdd2560290b70c46a86aee671 against 47a18a4bb44fb88dfa934e1aa3cfcd6a1ff6d047 are in! @Kha

- ✅ `build//instructions`: **-277.7G** (**-2.14%**)

**Large changes (1✅, 1🟥)**

- 🟥 `Init.Data.BitVec.Lemmas//instructions`: **+2.7G** (**+2.11%**)
- ✅ `build/module/Std.Tactic.BVDecide.Bitblast.BVExpr.Circuit.Lemmas.Expr//instructions`: **-10.7G** (**-8.47%**)

**Medium changes (19✅, 3🟥)**

Too many entries to display here. View the full report on radar instead.

**Small changes (1844✅, 46🟥)**

Too many entries to display here. View the full report on radar instead.

**Comment by adomani** [comment]
I haven't been able to look at this at a computer, but doesn't guard_msgs have a setting where it checks if s substring is present?  Maybe we can simply define a syntax before the command and check the find_syntax finds is as a test.

**Comment by Kha** [comment]
@adomani Hmm, I didn't see a substring option immediately in the docstring. It may be easier to just call the find_syntax implementation directly and do some checks on its output value rather than relying on the text message

**Comment by nomeata** [comment]
> @adomani Hmm, I didn't see a substring option immediately in the docstring. It may be easier to just call the find_syntax implementation directly and do some checks on its output value rather than relying on the text message

It was added in #11908.
